### PR TITLE
feat(api): create new endpoint import bom for component

### PR DESCRIPTION
**Pull request by tuannn2**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

### Description
_Endpoint import SBOM in component page._

Issue:  #1940

#### 1. API Prepare Import SBOM  -> _It will notify the user what components and releases are created when doing the import_

- Run command line: 
```
$ curl 'https://sw360.org/api/components/prepareImport/SBOM?type=SPDX' -i -X POST \
    -H 'Content-Type: multipart/form-data' \
    -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsic3czNjAtUkVTVC1BUEkiXSwidXNlcl9uYW1lIjoiYWRtaW5Ac3czNjAub3JnIiwic2NvcGUiOlsiYWxsIl0sImV4cCI6MTY4NDczMDM4MywiYXV0aG9yaXRpZXMiOlsiUkVBRCIsIldSSVRFIl0sImp0aSI6ImhBUzEySkJvNHNwUks3N21XaWxtcF9aa2Z4MCIsImNsaWVudF9pZCI6InRydXN0ZWQtc3czNjAtY2xpZW50In0.Lcl6gEALs55yC6_9VEJXRUCwIDDeUgS3N98nQftven53TihliRABdoP8_MmeNlmo4e8LMGJpl2IxpnD_Itl97tFT13x6IlKGG5nsPb3-l4k1wcjhHX6RH2IawUr-Pl0F8j5QVjtgxhD0Dff7_hADNdzIbHDuaMuGRAw0ePTxGuo3ehZKSL7bLlijJfJ4zyPUVP_Edv10zbf3CBj5B5EVfRhsTJKDyKGVMCbIHEXzDpDicZ68S53_8flhR5wafiAHYd5tu3XqkLwbaH8Q1CgL7bgvgT7AFYFKMJl6YoyVMWW3-wTcOxUcpPXzLwjbh5Tg0EAxMLbLxQ8_UK39g1lGTg' \
    -d 'file=@/bom.spdx.rdf'
```
- Expected Output
```
{
  "componentsName" : "Maven",
  "releasesName" : "Green Web 1.0.0"
}
```

#### 2. API Import SBOM -> _Import Components and Releases_ 

- Run command line: 
```
$ curl 'https://sw360.org/api/components/import/SBOM?type=SPDX' -i -X POST \
    -H 'Content-Type: multipart/form-data' \
    -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsic3czNjAtUkVTVC1BUEkiXSwidXNlcl9uYW1lIjoiYWRtaW5Ac3czNjAub3JnIiwic2NvcGUiOlsiYWxsIl0sImV4cCI6MTY4NDczMDM4MywiYXV0aG9yaXRpZXMiOlsiUkVBRCIsIldSSVRFIl0sImp0aSI6ImdmdDhiM0pNcm4yVVloaDZiWlg2cnZTajZVYyIsImNsaWVudF9pZCI6InRydXN0ZWQtc3czNjAtY2xpZW50In0.E1pbtB-wEyqezbjYTcIwyp7jzsitpp6vYeuOFiaLVKt5bT7N-xDsXgmJ7PGAZLMajrIWvoJwG2sJWQLX4NxhIG1oQgcdX9zS-M7xb-wFeBDY7RX0okjrhbVwng4kGoW1rAa5JbdDCKIAANru0h5V6T-Oo_lm8OOoQSTUtEkcLZV4yUln0vbGX7iaFxouTam356QYs8vpjAg0lRKpmPLf_PrBZA62BVqARAGYDl_F_6rpL7jlCgeeVwi9ww98aGnPgycH3E6gS4_KUve1Sco4xwBfkOwyrFvQywRe04JBps67kNWF7dBCozptGGvG24AbTvosnmMeTMiZN0bgwt8-gQ' \
    -d 'file=@/bom.spdx.rdf'
```
- Expected Output
```
{
  "name" : "Maven",
  "createdOn" : "2023-04-30",
  "componentType" : "SERVICE",
  "visbility" : "EVERYONE",
  "businessUnit" : "sw360 BA",
  "setBusinessUnit" : true,
  "setVisbility" : true,
  "_links" : {
    "self" : {
      "href" : "https://sw360.org/api/components/2222222"
    }
  },
  "_embedded" : {
    "createdBy" : {
      "email" : "admin@sw360.org",
      "deactivated" : false,
      "_links" : {
        "self" : {
          "href" : "https://sw360.org/api/users/byid/123456789"
        }
      }
    },
    "sw360:attachments" : [ {
      "filename" : "bom.spdx.rdf",
      "sha1" : "df90312312312534543544375345345383",
      "attachmentType" : "SBOM",
      "_links" : {
        "self" : {
          "href" : "https://sw360.org/api/attachments/3331231254"
        }
      }
    } ]
  }
}
```

### CODE REVIEW

**Comment by nikkuma7** [comment]
Test successful.
image

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java [hunk 1]
```diff
@@ -481,4 +484,72 @@ public ResponseEntity<CollectionModel<EntityModel>> getMyComponents(Pageable pag
...
+            throw new RuntimeException("failed to upload attachment", e);
+        }
+        ImportBomRequestPreparation importBomRequestPreparationResponse = handleImportBomRequestPreparation(importBomRequestPreparation);
+        HttpStatus status = importBomRequestPreparationResponse != null ? HttpStatus.OK : HttpStatus.BAD_REQUEST;
+        return new ResponseEntity<>(importBomRequestPreparationResponse, status);
+    }
+
+    private ImportBomRequestPreparation handleImportBomRequestPreparation(ImportBomRequestPreparation importBomRequestPreparation) {
+        ImportBomRequestPreparation importBomRequestPreparationResponse = new ImportBomRequestPreparation();
+        if (importBomRequestPreparation.isComponentDuplicate == true && importBomRequestPreparation.isReleaseDuplicate == true) {
```
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java:547]: Please avoid explicit checks for booleans to promote readability.

```suggestion
        if (importBomRequestPreparation.isComponentDuplicate && importBomRequestPreparation.isReleaseDuplicate) {
```

**Review by GMishx — CHANGES_REQUESTED** [review]
A minor change needed :-)

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java (hunk 1 above)
Comment (tuannn2) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java]: Thank you for your comment, @GMishx. 
I have updated.

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Pull request merged**